xarray/core/utils.py

```diff
@@ -274,22 +274,19 @@ def either_dict_or_kwargs(
     kw_kwargs: Mapping[str, T],
     func_name: str,
 ) -> Mapping[Hashable, T]:
-    if pos_kwargs is not None:
-        if not is_dict_like(pos_kwargs):
-            raise ValueError(
-                "the first argument to .%s must be a dictionary" % func_name
-            )
-        if kw_kwargs:
-            raise ValueError(
-                "cannot specify both keyword and positional "
-                "arguments to .%s" % func_name
-            )
-        return pos_kwargs
-    else:
+    if pos_kwargs is None:
         # Need an explicit cast to appease mypy due to invariance; see
         # https://github.com/python/mypy/issues/6228
         return cast(Mapping[Hashable, T], kw_kwargs)
 
+    if not is_dict_like(pos_kwargs):
+        raise ValueError("the first argument to .%s must be a dictionary" % func_name)
+    if kw_kwargs:
+        raise ValueError(
+            "cannot specify both keyword and positional " "arguments to .%s" % func_name
```

Comment:
```suggestion
            f"cannot specify both keyword and positional arguments to {func_name}"
```
dido concerning the `"."`